protocol/messenger_handler.go

```diff
@@ -1044,6 +1044,13 @@ func (m *Messenger) handleRetractContactRequest(state *ReceivedMessageState, con
 		return nil
 	}
 
+	m.allContacts.Store(contact.ID, contact)
+
+	if contact.Dismissed() {
+		// We dismissed this contact request locally already, no need to proceed
+		return nil
+	}
```

Comment:
Hmmmm, looking at `ContactRequestRetracted` implementation, which is called right above:
https://github.com/status-im/status-go/blob/2c48ce7ecf182770b36a28bf1b3e96ee8feb2097/protocol/contacts/contact.go#L305-L323

We already check the dismissed state there, maybe we should modify `ContactRequestRetracted` instead?